src/main/java/io/chazza/advancementapi/AdvancementAPI.java

```diff
@@ -238,14 +252,13 @@ public String toString() {
         }
     }
 
-
     //BEGIN CLASSES
     @Builder(builderMethodName = "hiddenbuilder")
     public static class Condition {
         protected String name;
-        protected JSONObject set;
+        protected JsonObject set;
```

Comment:
Here is a page http://www.minecraftforum.net/forums/minecraft-discussion/redstone-discussion-and/commands-command-blocks-and/2809368-1-12-custom-advancements-aka-achievements#Index

Not every Condition has a key. I think we have to differentiate for each condition which element can be used.
https://www.planetminecraft.com/blog/minecraft-custom-advancement-tutorial-1-12-snapshots/

[Example Array of items](http://image.prntscr.com/image/38dbc287d8934cbe960091da04ec6db6.png)

[Example parent breeding](http://image.prntscr.com/image/1dc02fcda3834cc58b64b3ef5a80a320.png)


`{
  "criteria": {
    "test": {
      "trigger": "minecraft:consume_item",
      "conditions": {
        "potion": { // currently this is wrong it has to be an array of items
          "item": "minecraft:apple",
          "amount": 1,
          "data": 0
        }
      }
    }
  },
  "display": {
    "icon": {
      "item": "minecraft:golden_apple"
    },
    "background": "minecraft:textures/gui/advancements/backgrounds/stone.png",
    "frame": "goal",
    "announce_to_chat": true,
    "show_toast": false,
    "hidden": false
  }
}`